packages/use-contractkit/src/use-contract-kit-methods.ts

```diff
@@ -42,7 +48,17 @@ export function useContractKitMethods(
         initialisedConnector.onNetworkChange?.((chainId) => {
           // TODO: We should probably throw an error if we can't find the new chainId
           const network = networks.find((n) => n.chainId === chainId);
-          network && dispatch('setNetwork', network);
+
+          if (network) {
+            dispatch('setNetwork', network);
+            initialisedConnector.updateKitWithNetwork &&
+              initialisedConnector
+                .updateKitWithNetwork(network)
+                .then(() => {
+                  dispatch('initialisedConnector', initialisedConnector);
+                })
+                .catch((e) => console.log(e));
```

Comment:
do we really just wanna ignore errors here?

Reply:
Good catch